src/search3_httpd.erl

```diff
@@ -0,0 +1,179 @@
+-module(search3_httpd).
+
+-export([handle_search_req/3]).
+
+-include("search3.hrl").
+-include_lib("couch/include/couch_db.hrl").
+
+-import(chttpd, [
+    send_method_not_allowed/2,
+    send_json/2,
+    send_json/3,
+    send_error/2
+]).
+
+handle_search_req(Req, Db, DDoc) ->
+    handle_search_req(Req, Db, DDoc, 0, 500).
+
+handle_search_req(#httpd{method=Method, path_parts=[_, _, _, _, IndexName]}=Req
+                  ,Db, DDoc, _, _)
+  when Method == 'GET'; Method == 'POST' ->
+    QueryArgs = #index_query_args{
+        include_docs = IncludeDocs
+    } = parse_index_params(Req, Db),
+    validate_search_restrictions(Db, DDoc, QueryArgs),
+    case search3_query:run_query(Db, DDoc, IndexName, QueryArgs) of
+        {Matches, Groups} ->
```

Comment:
distinguishing group- vs non-group- search results by the size of the tuple is brittle. can we get an atom in there? `{group_results, Matches, Groups}` or such?